### PR TITLE
#17738 Repro: "Native query editing" permission selector is inconsistent between "No self service" and "Block" permissions

### PR DESCRIPTION
**Pull request by nemanjaglumac**
### Status
PENDING REVIEW

### What does this PR accomplish?
- Reproduces #17738

### How to test this manually?
- `yarn test-cypress-open`
- `frontend/test/metabase/scenarios/admin/permissions/permissions.cy.spec.js`
- Replace `it.skip()` with `it.only()` to run the test in isolation
- The test should fail until the related issue is fixed

### Additional notes:
- Once the issue is fixed, please remove the `.skip` part (unskip the test completely)
- Make sure the test is passing and
- Merge it together with the fix

### Screenshots:
image

### CODE REVIEW

**Review by flamber — COMMENTED** [review]
I don't know what the behavior should be. Seems like it should probably be disabled (same for Granular), but someone needs to comment on the issue.

**Comment by nemanjaglumac** [comment]
As I understood from the bug bash, we want to go with the option that still enables you to edit SQL permissions. Let's wait.

**Comment by nemanjaglumac** [comment]
@brunobergher and @flamber updated the test according to [this comment](https://github.com/metabase/metabase/issues/17738#issuecomment-912657571).

File: frontend/test/metabase/scenarios/admin/permissions/permissions.cy.spec.js [hunk 1]
```diff
@@ -633,3 +654,17 @@ function assertPermissionTable(rows) {
     });
   });
 }
+
+/**
+ * @param {string} permission
+ * @param {boolean} isDisabled
```
Comment (gusaiani) [inline frontend/test/metabase/scenarios/admin/permissions/permissions.cy.spec.js:660]: Total nitpick, doc should probably state that function returns a boolean?

**Review by gusaiani — APPROVED** [review]
👍

File: frontend/test/metabase/scenarios/admin/permissions/permissions.cy.spec.js (hunk 1 above)
Comment (nemanjaglumac) [reply frontend/test/metabase/scenarios/admin/permissions/permissions.cy.spec.js:660]: I'm never sure how to document the return value when it actually returns the Cypress command (which will then resolve to a certain value, dom element, etc) :/

**Pull request merged**